From 1ccc0d8bcbb879aefe8fcba9ba23ba96eefdb103 Mon Sep 17 00:00:00 2001
From: Holger Weiss <holger@zedat.fu-berlin.de>
Date: Sun, 12 Oct 2014 19:44:35 +0200
Subject: [PATCH] XEP-0198: Set #state.conn field on session resume

Fixed in ejabberd 14.07+1. This bug causes ejabberdctl connected_users_info to
show "unknown" connections.

---
 src/ejabberd_c2s.erl | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

--- a/src/ejabberd_c2s.erl
+++ b/src/ejabberd_c2s.erl
@@ -2932,7 +2932,8 @@
 			      {auth_module, StateData#state.auth_module}],
 		      ejabberd_sm:open_session(NewSID, U, S, R,
 					       Priority, Info),
-		      {ok, StateData#state{sid = NewSID,
+		      {ok, StateData#state{conn = Conn,
+					   sid = NewSID,
 					   jid = OldStateData#state.jid,
 					   resource = OldStateData#state.resource,
 					   pres_t = OldStateData#state.pres_t,
